lib/dao.js

```diff
@@ -1183,8 +1191,9 @@ DataAccessObject._coerce = function (where) {
       }
     } else {
       if (val != null) {
-        if (!((operator === 'like' || operator === 'nlike') &&
-          val instanceof RegExp)) {
+        if (val instanceof RegExp) {
+          // do not coerce regex literals/objects
+        } else if (!((operator === 'like' || operator === 'nlike') && val instanceof RegExp)) {
```

Comment:
I see 3 cases:

1. if operator === 'regexp' and val is a regex literal/object, do nothing because we need the object to remain intact
2. if operator === like/nlike and val is regex object, convert to string because like/nlike uses string values
3. do nothing for any other data types

Is this correct?